### ISSUE TITLE
Group syncing toggle is turned off when creating a new mapping

### ISSUE THREAD

**Issue by albertoperdomo** [metabase/metabase#34914]
### Describe the bug

When creating the first mapping in group syncing, the toggle is turned off.
I don't get why we'd want to do this. But it seems weird. I spent some time troubleshooting on my own instance because of this.

Loom: https://www.loom.com/share/d6a09d82e0cd496f9dbd8fab04cc3a8d

Related: Also, it's weird that to persist the toggle state you need to save but mappings are auto-saved. But I am happy to open a separate issue for that.

### To Reproduce

1. With SSO working and group syncing off
2. Navigate to SSO settings
3. Turn group syncing off
4. Create mapping -> toggle for group syncing is turned off again

### Expected behavior

We shouldn't change the toggle setting, if we want to to do that (why?), at least warn the user before

### Information about your Metabase installation

```JSON
- Metabase v1.47.4
```

### Severity

Low

**Comment by npfitz** [comment]
@albertoperdomo I don't seem to be able to reproduce this on master. There is a chance this was addressed when the Webapp Admin team re-wrote the JWT form a few months ago (was merged in Nov). Any chance I can ask you to try to repro?

**Comment by albertoperdomo** [comment]
@npfitz Yes, I think it's been fixed. I just tested on 48.1 and can't reproduce.
